src/BSON/UTCDateTime.c

```diff
@@ -179,36 +179,40 @@ static PHP_METHOD(MongoDB_BSON_UTCDateTime, __construct)
 		return;
 	}
 
-	if (Z_TYPE_P(milliseconds) == IS_OBJECT) {
-		if (instanceof_function(Z_OBJCE_P(milliseconds), php_date_get_interface_ce())) {
-			php_phongo_utcdatetime_init_from_date(intern, Z_PHPDATE_P(milliseconds));
-		} else {
-			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "Expected instance of DateTimeInterface, %s given", ZSTR_VAL(Z_OBJCE_P(milliseconds)->name));
+	switch (Z_TYPE_P(milliseconds)) {
```

Comment:
I don't feel strongly, but I'd rather have extra `Z_TYPE_P()` calls than all of the extra indentation. Feel free to leave as-is, but I think the block scope in `IS_DOUBLE` needs a change.

Reply:
I extracted the logic for objects and doubles so the `switch` only defers to specific methods. This makes `construct` easier to understand at a glance and helps isolating the logic for the various input formats.

Reply:
Much more readable. Thanks!